src/spark/utils.py

```diff
@@ -122,3 +126,52 @@ def get_spark_session(
     Timer(int(timeout_sec), _stop_spark_session, [spark]).start()
 
     return spark
+
+
+def _detect_delimiter(sample: str) -> str or None:
+    """
+    Detect the delimiter of a CSV file from a sample string.
+
+    :param sample: A sample string from the CSV file
+
+    :return: The detected delimiter or None if it cannot be detected
+    """
+
+    try:
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(sample)
+        return dialect.delimiter
+    except Exception:
+        return None
+
+
+def read_csv(
+        spark: SparkSession,
+        path: str,
+        header: bool = True,
+        sep: str = None,
+        **kwargs
+) -> DataFrame:
+    """
+    Read a file in CSV format from minIO into a Spark DataFrame.
+
+    :param spark: The Spark session.
+    :param path: The minIO path to the CSV file.
```

Comment:
An example would help here given the parsing below

Reply:
👍

Reply:
:+1: